algorithms/sorting/tim_sort.py

```diff
@@ -0,0 +1,106 @@
+"""
+TimSort is a sorting algorithm based on Insertion Sort and Merge Sort.
+
+    Used in Python’s sorted() and sort().
+    First sort small pieces using Insertion Sort, then merges the pieces using merge of merge sort.
+    
+    Time Complexity:
+        O(n logn)
+"""
+# Python program to perform TimSort.  
+blockSize = 32 # Size of Subset to be sorted individually
+#You can Take any block size but I felt 32 to be fastest for me
+    
+# This function sorts array from left index to right index which is of size atmost blockSize  
+def insertionSort(arr, left, right):  
```

Comment:
```suggestion
def insertion_sort(arr, left, right):  
```

Reply:
What is the change here? Isnt it pep8 before change?

Reply:
I would say that the function names should be `lower_case_with_underscore` here. It is the convention for python function naming. You can check it more here: https://stackoverflow.com/questions/159720/what-is-the-naming-convention-in-python-for-variable-and-function-names